src/plugins/intel_gpu/src/plugin/transformations/transpose_fusion.cpp

```diff
@@ -53,13 +54,20 @@ bool has_optimized_version(const ov::Output<ov::Node>& output, bool supports_imm
     };
 
     const auto expected_dims_num = 4;
-    const auto original_dims_num = transpose_order.size();
-    if (original_dims_num < expected_dims_num) {
-        transpose_order.resize(expected_dims_num);
-        std::iota(transpose_order.begin() + original_dims_num, transpose_order.end(), original_dims_num);
-    }
 
-    if (!cldnn::one_of(transpose_order, allowed_orders))
+    std::vector<size_t> order(std::begin(transpose_order), std::end(transpose_order));
+    if (expected_dims_num > order.size()) {
+        size_t orders_to_add = expected_dims_num - order.size();
+        for (size_t i = 0; i < orders_to_add; ++i)
+            order.insert(order.begin(), i);
+        for (size_t i = orders_to_add; i < order.size(); ++i)
+            order[i] = order[i] + orders_to_add;
+    }
+    auto target_permute_order = order;
+    for (size_t i = 0; i < order.size(); ++i) {
+        target_permute_order[order[i]] = i;
+    }
+    if (!cldnn::one_of(target_permute_order, allowed_orders))
         return false;
```

Comment:
Could you please explain the motivation why non transposed dims should be added at the beginning but not to the end of transpose_order?

Reply:
@sshlyapn this was to align with the check in gemm onednn impl:
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/graph/impls/onednn/gemm_onednn.cpp#L170
or do you think we should have a common check for these 2 parts? to avoid future mismatching issue

Reply:
maybe have other scenario need to test, when support_immad = false and static shape, does ocl kernel support this transpose fusion, better we can have a check, or do you have any comment on this? @sshlyapn

Reply:
@songbell, thanks for clarification, I think the current version is okay
As for dynamic_shape and support_immad=false case, it should be supported; if I remember correctly, static shape for non-systolic platforms was disabled disabled because of performance reasons. @yeonbok please correct me if I'm wrong